### PR TITLE
machine-image: change base image to Ubuntu 22.04

### LINKED ISSUE #390: Update image base-version to ubuntu 22.04
Ubuntu 20.04 has less than 3 years of OS support remaining.

We should switch to Ubuntu 22.04 to reduce the need for OS upgrades in newly installed clusters.

### PR DESCRIPTION
**Pull request by benipeled**
This commit bumps the image base-version to Ubuntu 22.04

**Note:** this change is currently blocked by cloudformation issue [0], if it will not going to fix soon, we'll need to use a workaround to adjust the python modules paths as suggested on [0]

[0] https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1304

Fix #390

### CODE REVIEW

**Comment by yaronkaikov** [comment]
@benipeled The changes it self looks good, but we need the cloudformation working as well. so let's fix it here (another commit) and merge it together

**Comment by benipeled** [comment]
@yaronkaikov we're not going to fix it on our side but waiting for the fix on [0],
we have no rush with this change but if it will not fix before 5.2, we'll add a WA, for now let's leave it as draft

**Comment by fruch** [comment]
@benipeled even that SCT should be ready for 22.04, did you run any job with those images ? at least artifacts_ami ?

**Comment by benipeled** [comment]
@fruch yes and it failed https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/475/
we'll need to re-run verification once the cfn will be fixed but you're right, we can start adjusting the SCT

**Comment by fruch** [comment]
> @fruch yes and it failed https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/475/ we'll need to re-run verification once the cfn will be fixed but you're right, we can start adjusting the SCT

it failed cause of spot termination, both gce and azure passed cleanly.

**Comment by benipeled** [comment]
@fruch https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/477/ passed successfully,
let me know if there are any other requirements from SCT side for this change,

In any case, I'll rerun verification once we'll have the cfn fix

**Comment by benipeled** [comment]
We're no longer blocked by the cloudformation helper-scripts,
I verified this change with [1] - ubuntu 22.04.1 LTS & scylla 5.2.0~dev-0.20221107.a9063f9582f0 - it works as expected,

The target release for this change is 5.2, no early versions will release with ubuntu 22.04

@roydahan @fruch are there any other tests you want to run before merging this change?

PS> This PR should be merged together with https://github.com/scylladb/scylla-pkg/pull/3149

[1] https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/next-machine-image/518/

**Comment by roydahan** [comment]
If you can trigger AMI-test in releng-testing that triggers even one longevity job it will be great.

**Comment by benipeled** [comment]
> If you can trigger AMI-test in releng-testing that triggers even one longevity job it will be great.

AMI built successfully - https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/ami/1757/
artifacts-ami-test finished successfully -https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/artifacts/job/artifacts-ami-test/160/

A few longevity jobs are running...

**Review by yaronkaikov — CHANGES_REQUESTED** [review]

File: packer/build_image.sh
Comment (fruch) [reply packer/build_image.sh:31]: not clear why it was introduced in:
https://github.com/scylladb/scylla-machine-image/commit/4e1144a677bc5b0e171b37ec254ec31710a3c278

Comment (fruch) [reply packer/build_image.sh:31]: I agree that no part of the code uses it

**Comment by benipeled** [comment]
@roydahan @fruch longevity results

https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/longevity/job/longevity-100gb-4h-test/26/ - SUCCESS
https://jenkins.scylladb.com/job/scylla-master/job/releng-testing/job/longevity/job/longevity-10gb-3h-scylla-debug-test/25/ - FAILED on `test_custom_time` - timeout?

```
14:34:08  ======================================================================
14:34:08  ERROR: test_custom_time (longevity_test.LongevityTest)
14:34:08  Run cassandra-stress with params defined in data_dir/scylla.yaml
14:34:08  ----------------------------------------------------------------------
14:34:08  sdcm.sct_events.system.TestResultEvent: ================================= TEST RESULTS =================================
14:34:08  
14:34:08  ----- LAST ERROR EVENT -------------------------------------------------------
14:34:08  2022-11-09 12:23:12.033: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=552b4f43-a267-40a1-9a4f-eb6f3e0520e1, source=LongevityTest.test_custom_time (longevity_test.LongevityTest)() message=Traceback (most recent call last):
14:34:08  File "/home/ubuntu/scylla-cluster-tests/longevity_test.py", line 140, in test_custom_time
14:34:08  self.verify_stress_thread(cs_thread_pool=stress)
14:34:08  File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 1922, in verify_stress_thread
14:34:08  results, errors = cs_thread_pool.verify_results()
14:34:08  File "/home/ubuntu/scylla-cluster-tests/sdcm/stress_thread.py", line 285, in verify_results
14:34:08  for future in concurrent.futures.as_completed(self.results_futures, timeout=self.timeout):
14:34:08  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 241, in as_completed
14:34:08  raise TimeoutError(
14:34:08  concurrent.futures._base.TimeoutError: 1 (of 2) futures unfinished
```

**Comment by fruch** [comment]
@benipeled the debug job is disabled on master, so one is tracking its results.

So it's kind of pointless to keep triggering it.

It's basically says the Cassandra stress didn't finish in time we expected. which might be o.k. since it's using a debug build. But again we don't have enough information for this job history.

**Comment by fruch** [comment]
It's green light from my POV to do the switch.

**Review by syuu1228 — APPROVED** [review]
LGTM

**Review by yaronkaikov — APPROVED** [review]
LGTM

**Pull request merged**
